libphal/phal_sbe.C

```diff
@@ -370,5 +379,150 @@ void threadStopProc(struct pdbg_target *proc)
+ */
+void switchBootSeeprom(struct pdbg_target *proc)
+{
+	log(level::INFO, "reiplUsingBKPBootSeeprom on %s",
+	    pdbg_target_path(proc));
+}
+
+/**
+ * @brief Switch the measurment seeprom and REIPL using backup measurement
+ *        seeprom
+ * @param[in] proc processor target to operate on
+ */
+void switchMeasurementSeeprom(struct pdbg_target *proc)
+{
+	log(level::INFO, "switchMeasurementSeeprom on %s",
+	    pdbg_target_path(proc));
+}
+
+void validatePrimaryProcTgt(struct pdbg_target *proc)
+{
+	validatePrimaryProcTgt(proc);
+	if (!isPrimaryProc(proc)) {
+		log(level::ERROR,
+		    "validatePrimaryProcTgt: Target is not primary processor");
+		throw pdbgError_t(exception::PDBG_TARGET_INVALID);
+	}
+
+	isTgtPresent(proc);
+}
+
+void sbeErrorHandler(struct pdbg_target *proc)
+{
+	try {
+		log(level::INFO, "sbeErrorHandler on %s",
+		    pdbg_target_path(proc));
+
+		// Validate input target is processor target.
+		validatePrimaryProcTgt(proc);
+
+		enum sbe_state sbeState = getState(proc);
+
+		if (sbeState == SBE_STATE_DEBUG_MODE ||
+		    sbeState == SBE_STATE_FAILED) {
+			// The SBE on this processor is not in a state
+			// where chipOps can be issued. SBE Error handling
+			// will be skipped.
+			log(level::ERROR,
+			    "sbeErrorHandler sbe is not in DEBUG/FAILED"
+			    "state sbeState %d",
+			    pdbg_target_path(proc), sbeState);
+			throw sbeError_t(exception::SBE_INVALID_STATE);
+		}
+
+		// set state to debug mode to extract sbe rc
+		setState(proc, SBE_STATE_DEBUG_MODE);
+
+		// Execute SBE extract rc to set up sdb bit for pibmem dump to
+		// work
+		fapi2::ReturnCode fapiRc;
+		P10_EXTRACT_SBE_RC::RETURN_ACTION recovAction;
+
+		// p10_extract_sbe_rc is returning the error along with
+		// recovery action, so not checking the fapirc.
+		fapiRc = p10_extract_sbe_rc(proc, recovAction, true);
+		log(level::INFO,
+		    "sbeErrorHandler p10_extract_sbe_rc for proc=%s returned "
+		    "rc=0x%08X and SBE Recovery Action=0x%08X",
+		    pdbg_target_path(proc), fapiRc, recovAction);
+
+		// marking sbe state failed after p10_extract_sbe_rc
+		setState(proc, SBE_STATE_FAILED);
+
+		switch (recovAction) {
+		case P10_EXTRACT_SBE_RC::RETURN_ACTION::RESTART_SBE:
+		case P10_EXTRACT_SBE_RC::RETURN_ACTION::RESTART_CBS:
+			log(level::INFO,
+			    "sbeErrorHandler Recovery action RESTART_SBE/"
+			    "RESTART_CBS received on Proc %s",
+			    pdbg_target_path(proc));
+			(void)restartSBERCHandler(proc);
+			break;
+
+		case P10_EXTRACT_SBE_RC::RETURN_ACTION::REIPL_BKP_SEEPROM:
+		case P10_EXTRACT_SBE_RC::RETURN_ACTION::REIPL_UPD_SEEPROM:
+			log(level::INFO,
+			    "sbeErrorHandler Recovery action REIPL_BKP_SEEPROM/"
+			    "REIPL_UPD_SEEPROM received on Proc %s",
+			    recovAction, pdbg_target_path(proc));
+			(void)switchBootSeeprom(proc);
+			break;
+
+		case P10_EXTRACT_SBE_RC::RETURN_ACTION::REIPL_BKP_MSEEPROM:
+		case P10_EXTRACT_SBE_RC::RETURN_ACTION::REIPL_UPD_MSEEPROM:
```

Comment:
Same as the 480 line comment for measurement seeprom

Reply:
these error codes came from sbe and there was no difference of action in hwsv, so followed the same